diff -rc VirtualBox-6.1.22.orig/src/VBox/Devices/Audio/DrvHostAudioPulseAudioStubs.cpp VirtualBox-6.1.22/src/VBox/Devices/Audio/DrvHostAudioPulseAudioStubs.cpp
*** VirtualBox-6.1.22.orig/src/VBox/Devices/Audio/DrvHostAudioPulseAudioStubs.cpp	2021-04-28 09:27:05.000000000 -0700
--- VirtualBox-6.1.22/src/VBox/Devices/Audio/DrvHostAudioPulseAudioStubs.cpp	2021-05-20 14:10:59.988093502 -0700
***************
*** 333,339 ****
      rc = g_rc;
      AssertMsgReturn(rc == VERR_IPE_UNINITIALIZED_STATUS, ("g_rc=%Rrc\n", rc), rc);
  
!     rc = RTLdrLoadSystemEx(VBOX_PULSE_LIB, RTLDRLOAD_FLAGS_NO_UNLOAD, &hLib);
      if (RT_SUCCESS(rc))
      {
          unsigned i;
--- 333,339 ----
      rc = g_rc;
      AssertMsgReturn(rc == VERR_IPE_UNINITIALIZED_STATUS, ("g_rc=%Rrc\n", rc), rc);
  
!     rc = RTLdrLoad(VBOX_PULSE_LIB, &hLib); /* XXX: revert to 6.1.18 */ 
      if (RT_SUCCESS(rc))
      {
          unsigned i;
Only in VirtualBox-6.1.22.orig/src/VBox/Devices/Audio: .DrvHostAudioPulseAudioStubs.cpp.swp
